sympy/matrices/common.py

```diff
@@ -2377,23 +2393,25 @@ def __pow__(self, exp):
 
         return self.pow(exp)
 
-    def pow(self, exp, jordan=None, dotprodsimp=None):
+
+    def pow(self, exp, method="None"):
+
         """Return self**exp a scalar or symbol.
 
         Parameters
         ==========
 
-        jordan : bool, optional
-            If left as None then Jordan form exponentiation will be used under
-            certain conditions, True specifies that jordan_pow should always
-            be used if possible and False means it should not be used unless
-            it is the only way to calculate the power.
 
-        dotprodsimp : bool, optional
-            Specifies whether intermediate term algebraic simplification is used
-            during naive matrix power to control expression blowup and thus
-            speed up calculation. If the power is calculated using Jordan form
-            then this option has no effect. Default is on.
+        method : mulsimp, jordon, cayley
+            If jordon then Jordan form exponentiation will be used under
+            certain conditions.
```

Comment:
This is not how it should work. If the user specifies method='jordan' then it should always use the Jordan form method.

If the user does not specify a method (method=None not `"None"`) then we can decide what we think the appropriate method is and maybe in some cases that is the Jordan form method and in others it is something else.

The structure of this function should be:
```
# preliminaries e.g. check matrix is square

if method == 'meth1':
    # do meth1
elif method == 'meth2':
    # do meth2
...
elif method is None:
    # Decide heuristically which method to apply
    # including handling the diagonal case
else:
    raise TypeError('No such method')
```

Reply:
Is this okay now?